apps/ledger-live-desktop/src/renderer/drawers/Provider.tsx

```diff
@@ -1,29 +1,28 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
 import React, { useReducer, useEffect, useCallback, useState, useContext } from "react";
 import { DrawerProps as SideDrawerProps } from "~/renderer/components/SideDrawer";
 
+type ExtractProps<TComponentOrTProps> = TComponentOrTProps extends React.ComponentType<infer TProps>
+  ? TProps
+  : undefined;
+
```

Comment:
Very nice improvement ! 👍
One question, why the `OrTProps` in the naming of `TComponentOrTProps` ?